File: SkaiLab.Invoice.Models/Payroll.cs
using System;
using System.Collections.Generic;

namespace SkaiLab.Invoice.Models
{
    public class PayrollMonth
    {
        public long Id { get; set; }
        public string Month { get; set; }
        public decimal Total { get; set; }
        public decimal ExchangeRate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string OrganisationId { get; set; }
        public Currency Currency { get; set; }
    }
    public class PayrollMonthTaxSalary
    {

    }
    public class PayrollMonthTax: PayrollMonth
    {
        public decimal ChildOrSpouseAmount { get; set; }
        public decimal NoneResidentRate { get; set; }
        public decimal AdditionalBenefitsRate { get; set; }
        public List<TaxSalaryRange> TaxSalaryRanges { get; set; }
        public List<PayrollTax> Payrolls { get; set; }
       
        public Currency TaxCurrency { get; set; }
    }
    public class PayrollMonthNoneTax : PayrollMonth
    {
        public List<PayrollNoneTax> Payrolls { get; set; }
    }
    public class Payroll
    {
        public long Id { get; set; }
        public long EmployeeId { get; set; }
        public Employee Employee { get; set; }
        public DateTime Date { get; set; }
        public DateTime TransactionDate { get; set; }
        public decimal Total { get; set; }
        public Organsation Organsation { get; set; }
    }
    public class PayrollTax : Payroll
    {
        public decimal Salary { get; set; }
        public int? NumberOfChilds { get; set; }

        public decimal DeductSalary { get; set; }
        public decimal? OtherBenefit { get; set; }
        public decimal? OtherBenefitTaxDeduct { get; set; }

    }
    public class PayrollNoneTax : Payroll
    {
        public decimal Salary { get; set; }
        public decimal? OtherBenefit { get; set; }
    }
    public class TaxSalaryRange
    {
        public long Id { get; set; }
        public decimal FromAmount { get; set; }
        public decimal? ToAmount { get; set; }
        public double Rate { get; set; }
    }
}

File: SkaiLab.Invoice/ClientApp/src/app/models/payroll.ts
import { Currency } from "./currency";
import { Employee } from "./employee";

export class PayrollMonth{
    id:number;
    month:string;
    total:number;
    exchangeRate:number;
    startDate:Date;
    endDate:Date;
    organisationId:string;
    currency:Currency;
    constructor(){
        this.id=0;
        this.month="";
        this.exchangeRate=0;
        this.startDate=new Date();
        this.endDate=new Date();
        this.organisationId="";
        this.currency=new Currency();
    }
}
export class PayrollMonthTax extends PayrollMonth{
    childOrSpouseAmount:number;
    noneResidentRate:number;
    additionalBenefitsRate:number;
    taxSalaryRanges:TaxSalaryRange[];
    payrolls:PayrollTax[];
    taxCurrency:Currency;
    constructor(){
        super();
        this.childOrSpouseAmount=0;
        this.noneResidentRate=0;
        this.additionalBenefitsRate=0;
        this.taxSalaryRanges=[];
        this.payrolls=[];
        this.taxCurrency=new Currency();
    }
}
export class PayrollMonthNoneTax extends PayrollMonth{
    payrolls:PayrollNoneTax[];
    constructor(){
        super();
        this.payrolls=[];
    }
}
export class TaxSalaryRange{
    id:number;
    fromAmount:number;
    toAmount?:number;
    rate:number;
    constructor(){
        this.id=0;
        this.fromAmount=0;
        this.toAmount=0;
        this.rate=0;
    }

}

export class Payroll{
    id:number;
    employeeId:number;
    employee:Employee;
    date:Date;
    transactionDate:Date;
    total:number;
    constructor(){
        this.id=0;
        this.employeeId=0;
        this.employee=new Employee();
        this.date=new Date();
        this.transactionDate=new Date();
        this.total=0;
    }

}
export class PayrollNoneTax extends Payroll{
    salary:number;
    otherBenefit?:number;
    constructor(){
        super();
        this.salary=0;
        this.otherBenefit=0;
    }

}
export class PayrollTax extends Payroll{
    salary:number;
    numberOfChilds?:number;
    hasSpouse:boolean;
    deductSalary:number;
    otherBenefit?:number;
    otherBenefitTaxDeduct?:number;
    constructor(){
        super();
        this.salary=0;
        this.numberOfChilds=0;
        this.hasSpouse=false;
        this.deductSalary=0;
        this.otherBenefit=0;
        this.otherBenefitTaxDeduct=0;
    }

}